src/compiler/scala/tools/nsc/transform/CleanUp.scala

```diff
@@ -501,6 +501,66 @@ abstract class CleanUp extends Statics with Transform with ast.TreeDSL {
         reducingTransformListApply(rest.elems.length) {
           super.transform(localTyper.typedPos(tree.pos)(consed))
         }
+      //methods on Double
+      //new Predef.doubleToDouble(x).isNaN() -> java.lang.Double.isNaN(x)
+      //new Predef.doubleToDouble(x).isInfinite() -> java.lang.Double.isInfinity(x)
+      //methods on Float
+      //new Predef.float2Float(x).isNaN() -> java.lang.Double.isNaN(x)
+      //new Predef.float2Float(x).isInfinite() -> java.lang.Double.isInfinity(x)
+
+      //methods on Number
+      //new Predef.<convert>(x).byteValue() -> x.toByte()
+      //new Predef.<convert>(x).shortValue() -> x.toShort()
+      //new Predef.<convert>(x).intValue() -> x.toInt()
+      //new Predef.<convert>(x).longValue() -> x.toLong()
+      //new Predef.<convert>(x).floatValue() -> x.toFloat()
+      //new Predef.<convert>(x).doubleValue() -> x.toDouble()
+      //
+      // for each of the conversions
+      // double2Double
+      // float2Float
+      // byte2Byte
+      // short2Short
+      // char2Character
+      // int2Integer
+      // long2Long
+      // boolean2Boolean
+      //
+      case Apply(Select(Apply(boxing @ Select(qual, _), params), methodName), Nil)
+        if currentRun.runDefinitions.PreDef_primitives2Primitives.contains(boxing.symbol) &&
+          params.size == 1 &&
+          allFloatingMethodsToRewrite.contains(methodName) &&
+          treeInfo.isExprSafeToInline(qual) =>
+        val newTree =
+          if (doubleAndFloatRedirectMethods.contains(methodName)) {
+            val cls =
+              if (boxing.symbol == currentRun.runDefinitions.Predef_double2Double)
+                definitions.BoxedDoubleClass
+              else definitions.BoxedFloatClass
+
+            val targetMethod = cls.companionModule.info.decl(doubleAndFloatRedirectMethods(methodName))
+            gen.mkMethodCall(targetMethod, params)
+          } else {
+            gen.mkMethodCall(Select(params.head, javaNumberConversions(methodName)), Nil)
+          }
+        super.transform(localTyper.typedPos(tree.pos)(newTree))
+
+      //(x:Int).hashCode is transformed to scala.Int.box(x).hashCode()
+      //(x:Int).toString is transformed to scala.Int.box(x).toString()
+      //
+      //rewrite
+      // scala.Int.box(x).hashCode() ->  java.lang.Integer.hashCode(x)
+      // scala.Int.box(x).toString() ->  java.lang.Integer.toString(x)
+      // similarly for all primitive types
+      case Apply(Select(Apply(box @ Select(boxer, _), params), methodName), Nil)
+        if objectMethods.contains(methodName) &&
+          params.size == 1
+          && currentRun.runDefinitions.isBox(box.symbol)
```

Comment:
In principle this would be more correct

```suggestion
          params.size == 1 &&
          currentRun.runDefinitions.isBox(box.symbol) &&
          treeInfo.isExprSafeToInline(boxer)
```

Even if in reality that the backend currently drops the prefix; this is a bug that should be fixed in the backend.

```
scala> {println("hi"); Int}.box(1).hashCode
val res1: Int = 1
```